odl/tomo/geometry/conebeam.py

```diff
@@ -16,13 +16,457 @@
+            kwargs = {}
+        else:
+            kwargs = {'translation': translation}
+
+        return cls(apart, dpart, src_radius, det_radius, src_to_det,
+                   det_axis_init=det_axis, **kwargs)
+
+    @property
+    def src_radius(self):
+        """Source circle radius of this geometry."""
+        return self.__src_radius
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self.__det_radius
+
+    @property
+    def src_to_det_init(self):
+        """Initial source-to-detector unit vector."""
+        return self.__src_to_det_init
+
+    @property
+    def det_axis_init(self):
+        """Detector axis at angle 0."""
+        return self.detector.axis
+
+    def det_axis(self, angle):
+        """Return the detector axis at ``angle``."""
+        return self.rotation_matrix(angle).dot(self.det_axis_init)
+
+    @property
+    def angles(self):
+        """Discrete angles given in this geometry."""
+        return self.motion_grid.coord_vectors[0]
+
+    def src_position(self, angle):
+        """Return the source position at ``angle``.
+
+        For an angle ``phi``, the source position is given by::
+
+            src(phi) = translation +
+                       rot_matrix(phi) * (-src_rad * src_to_det_init)
+
+        where ``src_to_det_init`` is the initial unit vector pointing
+        from source to detector.
+
+        Parameters
+        ----------
+        angle : float
+            Rotation angle given in radians, must be contained in
+            this geometry's `motion_params`.
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape ``(2,)``
+            Source position corresponding to the given angle.
+
+        Examples
+        --------
+        With default arguments, the source starts at ``src_rad * (-e_y)``
+        and rotates to ``src_rad * e_x`` at 90 degrees:
+
+        >>> apart = odl.uniform_partition(0, 2 * np.pi, 10)
+        >>> dpart = odl.uniform_partition(-1, 1, 20)
+        >>> geom = FanFlatGeometry(apart, dpart, src_radius=2, det_radius=5)
+        >>> geom.src_position(0)
+        array([ 0., -2.])
+        >>> np.allclose(geom.src_position(np.pi / 2), [2, 0])
+        True
+        """
+        if self.check_bounds and not self.motion_params.contains_all(angle):
+            raise ValueError('`angle` {} not in the valid range {}'
+                             ''.format(angle, self.motion_params))
+
+        angle = np.array(angle, dtype=float, copy=False, ndmin=1)
+
+        # Initial vector from the rotation center to the source. It can be
+        # computed this way since source and detector are at maximum distance,
+        # i.e. the connecting line passes the origin.
+        center_to_src_init = -self.src_radius * self.src_to_det_init
+        pos_vec = (self.translation[None, :] +
+                   self.rotation_matrix(angle).dot(center_to_src_init))
+        return pos_vec.squeeze()
```

Comment:
If there is only one angle, we only want one vector back, not a `(1, ndim)` array. (I guess?)